src/main/java/org/breskul/actions/InsertAction.java

```diff
@@ -0,0 +1,97 @@
+package org.breskul.actions;
+
+import org.breskul.exception.BoboException;
+import org.breskul.util.AnnotationsHelper;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class InsertAction implements Action {
+    Connection connection;
+    Object object;
+
+    public InsertAction(Connection connection, Object object) {
+        this.connection = connection;
+        this.object = object;
+    }
+
+    @Override
+    public void execute() {
+        String tableName = AnnotationsHelper.getTableName(object);
+        var columns = AnnotationsHelper.getColumns(object);
+        String columnNames = String.join(",", columns.keySet());
+        String valuesPlaceholders = columns.keySet().stream().map(v -> "?").collect(Collectors.joining(","));
+        try {
+            var preparedStatement = connection.prepareStatement("INSERT INTO " + tableName
+                    + " (" + columnNames +") VALUES (" + valuesPlaceholders  + ")", Statement.RETURN_GENERATED_KEYS);
+            int i=1;
+            for (Map.Entry<String, Object> entry: columns.entrySet()){
+                var fieldClass = entry.getValue().getClass();
+                if (fieldClass.equals(String.class)) {
```

Comment:
I suppose we can get rid of such enormous count of if statements just by using preparedStatement.setObject()

Reply:
This is exactly what I want to test  when we add unit tests for multiple fields types. Will try suggested by you solution. (and keep previous as comment.)